app/src/main/java/com/mybus/MainActivity.java

```diff
@@ -31,9 +41,27 @@ protected void onCreate(Bundle savedInstanceState) {
     @Override
     public void onMapReady(GoogleMap googleMap) {
         mMap = googleMap;
-        // Add a marker in Mar Del Plata and move the camera
-        LatLng latLng = new LatLng(-37.979858, -57.589794);
-        mMap.addMarker(new MarkerOptions().position(latLng).title("Marker in Mar Del Plata"));
-        mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng, 12));
+        startLocationListening();
+        mMap.animateCamera(CameraUpdateFactory.zoomTo(DEFAULT_MAP_ZOOM));
+    }
+
+    public void setCurrentLocation(double lat, double lon) {
+        mMap.clear();
+        float currentZoom = mMap.getCameraPosition().zoom;
+        LatLng latLng = new LatLng(lat, lon);
+        mMap.addMarker(new MarkerOptions().position(latLng).title(CURRENT_LOCATION_MARKER));
+        mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng, currentZoom));
```

Comment:
CameraUpdateFactory.newLatLngZoom accepts only one param (latLng), no need to pass the current zoom if there is no change to it.